core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java

```diff
@@ -57,7 +64,9 @@
 
     private static final Pattern<FilterNode> PATTERN =
             filter().with(source().matching(
-                    tableScan().capturedAs(TABLE_SCAN)));
+                    tableScan().capturedAs(TABLE_SCAN)
+                            // avoid extra computations if table scan doesn't have any enforced predicate
+                            .matching(node -> !node.getEnforcedConstraint().isAll())));
```

Comment:
now that you cherried #10549, can you also apply the rename mentioned https://github.com/trinodb/trino/pull/10549/files#r783053560? thanks